vault/resource_database_secret_backend_connection_test.go

```diff
@@ -767,6 +767,54 @@ func TestAccDatabaseSecretBackendConnection_postgresql(t *testing.T) {
 	})
 }
 
+// This test makes sure that the DB connection resource is still
+// operational even when an external rotate root call is made to update
+// its credentials
+// Prerequisites: run the Postgres container found in docker-compose.yaml
+func TestAccDatabaseSecretBackendConnection_externalRotateRoot(t *testing.T) {
+	MaybeSkipDBTests(t, dbEnginePostgres)
+
+	username := "postgres"
+	password := "secret"
```

Comment:
Wouldn't that cause the Postgres container in CI to have it's credentials rotated? Thereby causing our other tests that use that same container to fail?

My thought was to have this test be a local only test where we will have to set up a Docker container locally — with the knowledge that the container's password will get rotated out after and it won't be usable after this test. The idea would be the container is only set up for that test and then destroyed once it's no longer usable by other tests

Curious to hear thoughts from folks! I don't think we have any rotate root tests in CI currently so this might be new areas that might need to be documented.

Reply:
Yeah I have it noted as a comment [here](https://github.com/hashicorp/terraform-provider-vault/pull/1876/files#diff-4fa185ba74c22b153981dd218fee9f1b4b962fe279ab789137724f251815fb05R776).

This would have to be revisited if we introduced new tests that also relied on the same PG container.

Reply:
I'll see if additional reruns causes any flakiness, and if so, I'll skip this test in CI.

Reply:
Alternatively we can introduce https://github.com/ory/dockertest, which it looks like we're already using in Vault.